File: examples/simple-stack-example-basic-java-view/src/main/java/com/zhuinden/navigationexampleview/core/navigation/BaseKey.java
package com.zhuinden.navigationexampleview.core.navigation;

import android.os.Parcelable;
import android.support.annotation.NonNull;

import com.zhuinden.simplestack.navigator.DefaultViewKey;
import com.zhuinden.simplestack.navigator.ViewChangeHandler;
import com.zhuinden.simplestack.navigator.changehandlers.FadeViewChangeHandler;

/**
 * Created by Owner on 2017. 06. 29..
 */

public abstract class BaseKey implements DefaultViewKey, Parcelable {
    @NonNull
    @Override
    public ViewChangeHandler viewChangeHandler() {
        return new FadeViewChangeHandler();
    }
}

File: examples/basic-examples/simple-stack-example-basic-kotlin-view/src/main/java/com/zhuinden/navigationexamplekotlinview/core/navigation/BaseKey.kt
package com.zhuinden.navigationexamplekotlinview.core.navigation

import android.os.Parcelable

import com.zhuinden.simplestack.navigator.DefaultViewKey
import com.zhuinden.simplestack.navigator.ViewChangeHandler
import com.zhuinden.simplestack.navigator.changehandlers.FadeViewChangeHandler

/**
 * Created by Owner on 2017. 06. 29..
 */

abstract class BaseKey : DefaultViewKey, Parcelable {
    override fun viewChangeHandler(): ViewChangeHandler = FadeViewChangeHandler()
}
